### ISSUE TITLE
Attempting to use SSL when Connect using SSL is unchecked

### ISSUE THREAD

**Issue by rsmck** [Sequel-Ace/Sequel-Ace#1351]
- Sequel Ace Version (including build number): 3.4.4 (20021), 3.4.5 (20025)
- Sequel Ace Source (App Store/GitHub/Homebrew): MAS, GitHub Binary
- macOS Version: 12.1 (21C52)
- Processor Type (Intel/Apple): Apple M1
- MySQL Version: Multiple, incl. 10.3.22-MariaDB-0+deb10u1-log
- macOS Localization: en-gb

**Is Issue Present in [Latest Beta](https://github.com/Sequel-Ace/Sequel-Ace/releases)?**
Yes

**Description**
Error `MySQL said: SSL connection error: error:1408F10B:SSL routines:ssl3_get_record:wrong version number` when connecting to multiple servers using TCP/IP or SSH even though "Connect using SSL" is unchecked. 

**Steps To Reproduce**
1. Create a new connection ensuring Connect Using SSL is unchecked (either as a favourite or using Quick Connect) 
2. Click Test Connection or Connect
3. Observe error above `MySQL said: SSL connection error....` which indicates SSL was used

**Expected Behaviour**
SequelAce should connect without using SSL

**Related Issues**
Likely related to closed issue #190 which seemed to be a narrower scope.

**Comment by Jason-Morcos** [comment]
Can you share the full connection error log (as a file attachment, not pasted text)?
I would prefer to always attempt all connections via SSL and then only fall back on not using SSL if the connection specifically cannot connect via SSL. I have some connections that don't connect via SSL and are currently working with fallback. I'd like to see from the full connection log to see if we try to fallback without SSL after the SSL connection fails

**Comment by Jason-Morcos** [comment]
This may also be useful - https://dev.mysql.com/doc/refman/5.7/en/using-encrypted-connections.html

**Comment by armetiz** [comment]
I have the same issue using latest Sequel ACE 3.4.5, with a MySQL server 5.7.19 without SSL.
The problem does not exists on Sequel Pro latest version.

I have the following favorites :

Not working : Engine version 5.7.19 - Certificate authority rds-ca-2015
Working : Engine version 5.7.21 - Certificate authority rds-ca-2015
Working : Engine version 5.7.19- Certificate authority rds-ca-2019


How can I help? How to get the "full connection error log"?

**Comment by rsmck** [comment]
Hi @Jason-Morcos sorry I didn't notice your reply until now. If "Connect using SSL" is not ticked it should not attempt to connect using SSL imho, if that is the intended behaviour then the tickbox shouldn't be present at all as it presumably doesn't serve a purpose? 

I'm not sure how to get a detailed connection log from Sequel Ace? (I can't see an option in the UI nor is it logged to Console

**Comment by armetiz** [comment]
@Jason-Morcos could you please tell us how to get connection log from Sequel Ace? 

Thank you.

**Comment by Jason-Morcos** [comment]
The error dialog that pops up when failing to connect should show a button to view the full log. Does it not show one in this case? Can someone screenshot that error dialog?

**Comment by armetiz** [comment]
I do not have any button to view the full log from the error dialog.

![sequel-ace-ssl-wrong-version](https://user-images.githubusercontent.com/1119250/151860478-61973fa7-840e-4775-8ef8-668a6128e7f9.png)

**Comment by rsmck** [comment]
Me neither, just the error above.

**Comment by rsmck** [comment]
This is still present in 3.5.0 (20028) Beta 2 

```
Unable to connect to host 127.0.0.1, or the request timed out.

Be sure that the address is correct and that you have the necessary privileges, or try increasing the connection timeout (currently 10 seconds).

MySQL said: SSL connection error: error:1408F10B:SSL routines:ssl3_get_record:wrong version number
```

No other log options

**Comment by Jason-Morcos** [comment]
If anyone who is experiencing this issue can build the app locally and test, I'd love to know if this fixes the issue. I haven't been able to reproduce it myself
https://github.com/Sequel-Ace/Sequel-Ace/pull/1381

**Comment by armetiz** [comment]
Do you have any documentation to build the application from source?

**Comment by rsmck** [comment]
Unfortunately haven't built from source #1381 doesn't seem to fix this.  I still amn't sure how to get more debug information from it during the connection process all I can see in the Xcode output is;

```
2022-02-09 12:51:18.618795+0000 Sequel Ace[12275:237918] -[SPMySQLConnection connect]:268: connect
2022-02-09 12:51:18.618908+0000 Sequel Ace[12275:237918] -[SPMySQLConnection(PrivateAPI) _connect]:532: _connect
2022-02-09 12:51:18.730980+0000 Sequel Ace[12275:237918] -[SPMySQLConnection(PrivateAPI) _connect]:559: !mySQLConnection, unlock
```

**Comment by armetiz** [comment]
awesome! I will wait for the official release.

good job @Jason-Morcos , thanks ;)

**Comment by Jason-Morcos** [comment]
The fix didn't make it in before we closed 3.5.0, but it'll be included in the next release (presumably 3.5.1!). Will release a beta once we release 3.5.0 officially
